src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -2,22 +1,65 @@
+import { Dispatch, SetStateAction } from 'react';
+import { Filter } from '../../App';
+import { useDispatch } from 'react-redux';
+import { filterSlice } from '../../features/filter';
+
+type Props = {
+  filterType: Filter;
+  query: string;
+  onFilter: Dispatch<SetStateAction<Filter>>;
+  onQuery: Dispatch<SetStateAction<string>>;
+  onResetQuery: () => void;
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  filterType,
+  query,
+  onFilter,
+  onQuery,
+  onResetQuery,
+}) => {
+  const dispatch = useDispatch();
 
-export const TodoFilter: React.FC = () => {
   return (
-    <form
-      className="field has-addons"
-      onSubmit={event => event.preventDefault()}
-    >
+    <form className="field has-addons">
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
-            <option value="all">All</option>
-            <option value="active">Active</option>
-            <option value="completed">Completed</option>
+          <select
+            data-cy="statusSelect"
+            value={filterType}
+            onChange={event => {
+              onFilter(event.target.value as Filter);
+              dispatch(filterSlice.actions.toggleFilter(event.target.value));
+            }}
+          >
+            <option
+              onClick={() =>
+                dispatch(filterSlice.actions.toggleFilter(Filter.All))
+              }
+              value={Filter.All}
+            >
+              All
+            </option>
+            <option
+              onClick={() =>
+                dispatch(filterSlice.actions.toggleFilter(Filter.Active))
+              }
```

Comment:
Similarly, using `onClick` handlers on `option` elements is not recommended. Consider removing these handlers as the `onChange` event on the `select` element already handles the dispatch logic.